AK/NumberFormat.cpp

```diff
@@ -110,25 +111,28 @@ DeprecatedString human_readable_digital_time(i64 time_in_seconds)
     return builder.to_deprecated_string();
 }
 
-DeprecatedString human_readable_integer(i64 number)
+DeprecatedString human_readable_integer(const i64 number)
 {
-    bool is_negative = false;
-    if (number < 0) {
-        is_negative = true;
-        number *= -1;
-    }
-
-    Vector<u64> separated_pieces;
+    const u64 abs_number = abs(number);
+    // Handle the edge case of number being 0 as log10 is undefined for it.
+    const u8 digit_count = ceil(abs_number > 0 ? log10(static_cast<long double>(abs_number)) : 0) + 1;
+    const u8 section_count = (digit_count - 1) / 3 + 1;
+
+    Vector<u16> separated_sections;
+    separated_sections.ensure_capacity(section_count);
+    u64 remainder = abs_number;
```

Comment:
Having number `const` is nice, but it forces creating a variable here.
That's not an issue, but we can take advantage of it by reducing the scope of `remainder` using a for loop instead.

Reply:
A for loop would not handle number == 0 correctly unlike this do-while though.

Reply:
I guess we could make a for loop with the condition changed to `remainder > 0 || separated_sections.is_empty()` if that's preferred.

Reply:
Doesn't matter much.